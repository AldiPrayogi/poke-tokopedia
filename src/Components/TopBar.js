/** @jsxImportSource @emotion/react */

import {ContainerGithubCSS,ContainerPokeballCSS, TopBarCSS, NavigationButtonCSS} from '../Styling/ComponentCSS';
import { useHistory } from 'react-router-dom'
import React from "react";

export const TopBar = () => {
    const history = useHistory();

    const handlePokeballClick =  () => {
        history.push('/');
    }

    const handleListClick =  () => {
        history.push('/pokemon-list');
    }

    const handlePokedexClick = () => {
        history.push('/pokedex');
    }

    const handleHomeClick = () => {
        history.push('/');
    }

    return(
        <div css={TopBarCSS}>
            <div css={ContainerPokeballCSS}>
                <a onClick={handlePokeballClick}><img src='/images/PinClipart.com_pokeball-clip-art_843046.png' alt='PokeBall' width='170px'/></a>
                <h2>GOTTA CATCH 'EM ALL!</h2>
            </div>
            <div css={ContainerGithubCSS}>
                <a target="_blank" href='https://github.com/AldiPrayogi/poke-tokopedia' rel='noreferrer'>GitHub</a>
            </div>
                <div css={NavigationButtonCSS}>
                    <a onClick={handleListClick}>Pokemon List</a>
                </div>
                <div css={NavigationButtonCSS}>
                    <a onClick={handleHomeClick} style={{margin: "2vw"}}>Home</a>
                </div>
                <div css={NavigationButtonCSS}>
                    <a onClick={handlePokedexClick}>Your Pokedex</a>
                </div>
        </div>
    );
}

