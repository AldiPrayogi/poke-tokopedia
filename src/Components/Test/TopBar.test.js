import {TopBar} from "../TopBar";
import {shallow} from 'enzyme';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockHistoryPush,
    }),
}));

afterEach(() =>{
    jest.clearAllMocks();
})

afterAll(() => {
    jest.clearAllMocks();
})

describe('TopBar', () => {
    const wrapper = shallow(<TopBar/>);

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe('handlePokeballClick', () => {
        it('should call useHistory.push with / when pokeball button is clicked', () => {
            const pokeBallButton = wrapper.find('.pokeball-button');

            pokeBallButton.simulate('click');

            expect(mockHistoryPush).toHaveBeenCalledWith('/');
        });

        it('should call useHistory.push with / when home button is clicked', () => {
            const homeButton = wrapper.find('.home-button');

            homeButton.simulate('click');

            expect(mockHistoryPush).toHaveBeenCalledWith('/');
        });
    });

    describe('handleListClick', () => {
        it('should call useHistory.push with /pokemon-list when pokemon list button is clicked', () => {
            const pokemonListButton = wrapper.find('.pokemon-list-button');

            pokemonListButton.simulate('click');

            expect(mockHistoryPush).toHaveBeenCalledWith('/pokemon-list');
        });
    });

    describe('handlePokedexClick', () => {
        it('should call useHistory.push with /pokedex when pokedex button is clicked', () => {
            const pokedexButton = wrapper.find('.pokedex-button');

            pokedexButton.simulate('click');

            expect(mockHistoryPush).toHaveBeenCalledWith('/pokedex');
        });
    });


});