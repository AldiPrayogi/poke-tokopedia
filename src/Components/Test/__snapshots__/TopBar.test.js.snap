// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should render correctly 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="TopBar"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  css={
    Object {
      "map": undefined,
      "name": "114om3",
      "next": undefined,
      "styles": "
    color: black;
    width: 100%;
    border: 0px;
    padding: auto;
    min-width: 200px;
    height: auto;
    text-align: center;
    background-image: linear-gradient(75deg, #bd4949, #7b2916);
",
      "toString": [Function],
    }
  }
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="TopBar"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "map": undefined,
        "name": "1ys08fa",
        "next": undefined,
        "styles": "
    width: 100%;
    padding: 6vh 0 2vh 0;
    img {
        display: block;
        margin-left: auto;
        margin-right: auto;
    }
    h2{
        text-align: center;
        font-style: italic;
        font-weight: bold;
        font-size: 6.5vh;
        padding-top: 5vh;
        margin: auto;
        color: #f0c44d;
        font-family: Verdana, sans-serif;
    }
",
        "toString": [Function],
      }
    }
  >
    <a
      className="pokeball-button"
      onClick={[Function]}
    >
      <img
        alt="PokeBall"
        src="/images/PinClipart.com_pokeball-clip-art_843046.png"
        width="170px"
      />
    </a>
    <h2>
      GOTTA CATCH 'EM ALL!
    </h2>
  </EmotionCssPropInternal>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="TopBar"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "map": undefined,
        "name": "4ddtbo",
        "next": undefined,
        "styles": "
    padding: 5vh 0 2vh 0;
    text-align: center;
    a {
        background-color: #222224;
        border-radius: 2vh;
        border-color: #f0c44d;
        border-style: solid;
        border-width: 1px;
        color: #f0c44d;
        padding: 2.5vh;
        font-weight: bold;
        font-size: 1.8vh;
    }
",
        "toString": [Function],
      }
    }
  >
    <a
      className="github-button"
      href="https://github.com/AldiPrayogi/poke-tokopedia"
      rel="noreferrer"
      target="_blank"
    >
      GitHub
    </a>
  </EmotionCssPropInternal>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="TopBar"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "map": undefined,
        "name": "n4tp1j",
        "next": undefined,
        "styles": "
     padding: 10vh 0 10vh 0;
     border: auto;
     display: inline-block;
     min-width: 60px;
     a {
         background-color: #222224;
         border-radius: 2vh;
         border-color: #f0c44d;
         border-style: solid;
         border-width: 1px;
         color: #f0c44d;
         padding: 2.5vh;
         font-weight: bold;
         font-size: 1.2vh;
     }
",
        "toString": [Function],
      }
    }
  >
    <a
      className="pokemon-list-button"
      onClick={[Function]}
    >
      Pokemon List
    </a>
  </EmotionCssPropInternal>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="TopBar"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "map": undefined,
        "name": "n4tp1j",
        "next": undefined,
        "styles": "
     padding: 10vh 0 10vh 0;
     border: auto;
     display: inline-block;
     min-width: 60px;
     a {
         background-color: #222224;
         border-radius: 2vh;
         border-color: #f0c44d;
         border-style: solid;
         border-width: 1px;
         color: #f0c44d;
         padding: 2.5vh;
         font-weight: bold;
         font-size: 1.2vh;
     }
",
        "toString": [Function],
      }
    }
  >
    <a
      className="home-button"
      onClick={[Function]}
      style={
        Object {
          "margin": "2vw",
        }
      }
    >
      Home
    </a>
  </EmotionCssPropInternal>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="TopBar"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "map": undefined,
        "name": "n4tp1j",
        "next": undefined,
        "styles": "
     padding: 10vh 0 10vh 0;
     border: auto;
     display: inline-block;
     min-width: 60px;
     a {
         background-color: #222224;
         border-radius: 2vh;
         border-color: #f0c44d;
         border-style: solid;
         border-width: 1px;
         color: #f0c44d;
         padding: 2.5vh;
         font-weight: bold;
         font-size: 1.2vh;
     }
",
        "toString": [Function],
      }
    }
  >
    <a
      className="pokedex-button"
      onClick={[Function]}
    >
      Your Pokedex
    </a>
  </EmotionCssPropInternal>
</EmotionCssPropInternal>
`;
